chain-core/src/property.rs

```diff
@@ -35,6 +35,21 @@
 //! is selected to write a block in the chain.
 //!
 
+use std::hash::Hash;
+
+/// Trait identifying the block identifier type.
+pub trait BlockId: Eq + Hash {}
+
+/// Trait identifying the block date type.
+pub trait BlockDate: Eq + Ord {}
+
+/// Trait identifying the transaction identifier type.
+pub trait TxId: Eq + Hash {}
```

Comment:
I would rather like to see `TransactionId` here. I know this was not the norm before but being more explicit to avoid confusion would help.